### PR TITLE
Refactor/sweep runner 2 cleanup

### PR DESCRIPTION
**Pull request by SerialVelocity**
Bit of clean-up and refactoring

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/734)

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java [hunk 1]
```diff
@@ -229,18 +232,19 @@ private boolean isLatestValueEmpty(Cell cell, PeekingIterator<RowResult<Value>>
...
-        SortedSet<Long> committedTimestampsToSweep = Sets.newTreeSet();
+    private SweepTimestampsAndSentinels getTimestampsToSweep(
+            Cell cell,
+            Collection<Long> startTimestamps,
+            LoadingCache<Long, Long> startTsToCommitTs,
+            long sweepTimestamp,
+            boolean sweepLastCommitted,
+            StrategySweeper strategySweeper) {
+        Set<Long> uncommittedTimestamps = new HashSet<>();
+        SortedSet<Long> committedTimestampsToSweep = new TreeSet<>();
```
Comment (hsaraogi) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java]: I feel the previous indent was more readable.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTimestampsAndSentinels.java [hunk 2]
```diff
@@ -23,13 +23,13 @@
...
     public static SweepTimestampsAndSentinels of(Set<Long> timestamps, Set<Cell> sentinelsToAdd) {
         return ImmutableSweepTimestampsAndSentinels.builder()
                 .timestamps(timestamps)
                 .sentinelsToAdd(sentinelsToAdd)
                 .build();
     }
+
+    public abstract Set<Long> timestamps();
+    public abstract Set<Cell> sentinelsToAdd();
 }
```
Comment (hsaraogi) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTimestampsAndSentinels.java]: Any reason for the shift?

File: atlasdb-impl-shared/src/test/java/com.palantir.atlasdb.sweep/SweepTaskRunnerImplTest.java [hunk 3]
```diff
@@ -244,35 +271,20 @@ public void thorough_getTimestampsToSweep_oneTransaction_emptyValue_returnsIt()
...
-        return timestampsPerRow;
-    }
-
-    private Set<Cell> getSentinelsToAdd(SweepStrategySweeper sweeper) {
-        long sweepTimestampHigherThanCommitTimestamp = HIGH_COMMIT_TS + 1;
-        Multimap<Cell, Long> timestampsPerRow = twoCommittedTimestampsForSingleCell();
+        when(mockTransactionService.get(timestampsPerRow.values()))
+                .thenReturn(ImmutableMap.of(
+                        LOW_START_TS, LOW_COMMIT_TS,
+                        HIGH_START_TS, HIGH_COMMIT_TS));
```
Comment (hsaraogi) [inline atlasdb-impl-shared/src/test/java/com.palantir.atlasdb.sweep/SweepTaskRunnerImplTest.java]: thenReturn(timestampsPerRow)

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/StrategySweeper.java [hunk 4]
```diff
@@ -23,7 +23,7 @@
 import com.palantir.atlasdb.keyvalue.api.Value;
 import com.palantir.common.base.ClosableIterator;
 
-public interface SweepStrategySweeper {
+public interface StrategySweeper {
```
Comment (gbrova) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/StrategySweeper.java]: Why not `SweeperStrategy`?

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java [hunk 5]
```diff
@@ -199,7 +194,10 @@ SweepCellsAndSentinels getStartTimestampsPerRowToSweep(
         LoadingCache<Long, Long> startTsToCommitTs = CacheBuilder.newBuilder()
                 .build(new StartTsToCommitTsCacheLoader(transactionService));
 
-        startTsToCommitTs.putAll(transactionService.get(startTimestampsPerCell.values()));
+        // Needed because calling transactionService.get(<EMPTY>) is weird (it logs that it is empty too).
+        if(startTimestampsPerCell.isEmpty()) {
```
Comment (gbrova) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java]: Shouldn't this be if _not_ empty?

Comment (gbrova) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java]: At least the build failed, so the tests are catching this.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java (hunk 1 above)
Comment (SerialVelocity) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java]: The problem with the other indentation is that it changes whenver the return type, method name, etc changes so you end up with massive diffs. indentation of 8 is what is used instead usually.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTimestampsAndSentinels.java (hunk 2 above)
Comment (SerialVelocity) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTimestampsAndSentinels.java]: constructors are usually at the top of files. Having them always in the same place tends to make the files much easier to skim through

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/StrategySweeper.java (hunk 4 above)
Comment (SerialVelocity) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/StrategySweeper.java]: Can you think of an even better name? We already have `SweepStrategy` so `SweeperStrategy` is wayyy too close to that for my liking. (But I agree that both `SweepStrategySweeper` and `StrategySweeper` are also awful)

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java (hunk 5 above)
Comment (SerialVelocity) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java]: Thank you ^^ I'll add a test xD

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java [hunk 6]
```diff
@@ -124,10 +124,6 @@ public SweepResults run(TableReference tableRef, int batchSize, @Nullable byte[]
         // (2) force old writers to retry (note that we must roll back any uncommitted transactions that
         //     we encounter
         SweepStrategy sweepStrategy = sweepStrategyManager.get().getOrDefault(tableRef, SweepStrategy.CONSERVATIVE);
-        if (sweepStrategy == SweepStrategy.NOTHING) {
```
Comment (gbrova) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java]: Is there any performance implication to removing this shortcircuit?  (I guess it doesn't matter if nothing important is using SweepStrategy.NOTHING)

Comment (SerialVelocity) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java]: Nope, just went through the code and it's a linear path. Shouldn't really slow anything down.

File: atlasdb-impl-shared/src/test/java/com.palantir.atlasdb.sweep/SweepTaskRunnerImplTest.java (hunk 3 above)
Comment (SerialVelocity) [inline atlasdb-impl-shared/src/test/java/com.palantir.atlasdb.sweep/SweepTaskRunnerImplTest.java]: ? Why? timestampsPerRow is START_TS, this is COMMIT_TS

**Pull request merged**